dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/persistence/datastore/DataStoreCallback.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.persistence.datastore
+
+/**
+ * Callback for asynchronous operations on the datastore.
+ */
+interface DataStoreCallback<T : Any> {
+
+    /**
+     * Called on successfully fetching data from the datastore.
+     *
+     * @param dataStoreContent contains the datastore data and version.
+     */
+    fun onSuccess(dataStoreContent: DataStoreContent<T>)
+
+    /**
+     * Called when an exception occurred getting data from the datastore.
+     */
+    fun onFailure()
+
+    /**
+     * Called when no data is found for the given key.
+     */
+    fun onNoData()
```

Comment:
side note: maybe we can reduce the number of methods in this interface by doing argument of `onSuccess` nullable and maybe renaming it to `onResult` or something like that, then we won't need this method, because no data is also successful result in some sense.